main.c

```diff
@@ -67,21 +115,116 @@ void main(void) {                       // Función principal
     max7219_set_scan_limit(MAX_DISPLAY_0, 7);
     max7219_set_intensity (MAX_DISPLAY_0, 15);
     max7219_clear_display(MAX_DISPLAY_0);
-    
-    max7219_set_led(MAX_DISPLAY_0, 5, 5, LED_ON);
+           
+//    printf("Este es un dado electronico\r\n");
+      printf("Este es un programa para dibujar\r\n");
     
     while(1) {                          // Super loop
         // Ver este link: https://pbs.twimg.com/media/BafQje7CcAAN5en.jpg
         
-        if( PIN_TEC1 == 0 ) {           // Espero que se presione la TEC1
-            __delay_ms(40);             // Delay antirrebote
+// Programa para dibujar             
+        
+//        max7219_clear_display(MAX_DISPLAY_0);
+//        max7219_set_led(MAX_DISPLAY_0, cursor.x, cursor.y, LED_ON);
+//        __delay_ms(1);
+//        max7219_set_led(MAX_DISPLAY_0, led_fijo_x, led_fijo_y, LED_ON);
+////        max7219_set_led(MAX_DISPLAY_0, cursor.x, cursor.y, LED_OFF);
+////        __delay_ms(125);
+//        
+//        if( uart_rx_byte( &dato_recibido ) ) {
+//            if( dato_recibido == 'w'){
+//                  cursor.y --;            
+//            if(cursor.y == -1 )
+//            cursor.y = 7;
+//            }
+//            if( dato_recibido == 'a'){
+//                  cursor.x --;            
+//            if(cursor.x == -1 )
+//            cursor.x = 7;
+//            }
+//            if( dato_recibido == 's'){
+//                  cursor.y ++;            
+//            if(cursor.y == 8 )
+//            cursor.y = 0;
+//            }
+//            if( dato_recibido == 'd'){
+//                  cursor.x ++;            
+//            if(cursor.x == 8 )
+//            cursor.x = 0;
+//            }
+//            if ( dato_recibido == 'f') {    
+//                led_fijo_x = cursor.x;
+//                led_fijo_y = cursor.y;
+//                    
+//                          
+//            }
+//         }
             
-            PIN_LED1 = !PIN_LED1;       // Toggle LED1
-
-            while( PIN_TEC1 == 0 );     // Espero que se suelte la TEC1
-            __delay_ms(40);             // Delay antirrebote
+//    }
+//        
+        
+        
+// Programa ball        
+        
+        max7219_clear_display(MAX_DISPLAY_0);
+        // Dibujo la pelota
+        max7219_set_led(MAX_DISPLAY_0, ball.x, ball.y, LED_ON);
+        // Dibujo la paleta
+        for(i = 0 ; i < line.size ; i++)
+        max7219_set_led(MAX_DISPLAY_0, line.y + i, line.x, LED_ON);
+        
+        
+        __delay_ms(100);
+        
+        // Encuestar teclas
+        if( PIN_TEC1 == 0 && line.y > 0){
+            line.y --;
+        }
+        if( PIN_TEC2 == 0 && line.y + i <= 7){
```

Comment:
Acá debería tenerse en cuenta el tamaño, antes funcionaba igual porque `i` es igual a `line.size` después del `for`.
Lo correcto es lo siguiente igual:

```suggestion
        if( PIN_TEC2 == 0 && line.y + line.size <= 7){
```